core/src/main/java/com/novoda/noplayer/internal/exoplayer/SkippedAdverts.java

```diff
@@ -43,15 +43,15 @@ static AdPlaybackState markAllNonPlayedAdvertsAsSkipped(List<AdvertBreak> advert
     static AdPlaybackState markAllPastAvailableAdvertsAsSkipped(long currentPositionInMillis,
                                                                 List<AdvertBreak> advertBreaks,
                                                                 AdPlaybackState adPlaybackState) {
-        AdPlaybackState adPlaybackStateWithSkippedAdGroups = adPlaybackState;
+        AdPlaybackState updatedPlaybackState = adPlaybackState;
         for (int i = advertBreaks.size() - 1; i >= 0; i--) {
             AdvertBreak advertBreak = advertBreaks.get(i);
-            AdPlaybackState.AdGroup adGroup = adPlaybackState.adGroups[i];
```

Comment:
here we were always modifying the initial state so we would only mark the last accessed group as skipped